### PR TITLE
WIP Endpoints

### PR DESCRIPTION
**Pull request by JonathanSR**
Working on finishing API endpoints
update foods endpoint is in progress, rest of API endpoints are complete

### CODE REVIEW

**Comment by JonathanSR** [comment]
Finishes api food endpoints
testing needed for 
delete api/foods/:id
update api/foods/:id

@case-eee @neight-allen @valerietrudell

File: lib/controllers/foods-controller.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  });
+};
+
+
+function create (request, response) {
+  let name = request.body.food_name
+  let calories = request.body.calories
+
+  if(name == null) {
```
Comment (case-eee) [inline lib/controllers/foods-controller.js:34]: i think you can do the same here as you do with calories on line 35. `if(!name)`

File: lib/controllers/foods-controller.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      }); 
+    } else {
+      Food.create(name, calories).then((data) => {
+        response.status(201).json(data)
+      });
+    };
+  };
+
+function destroy (request, response){
+Food.find(request.params.id)
```
Comment (case-eee) [inline lib/controllers/foods-controller.js:50]: watch your spacing here. this should be indented

File: lib/controllers/foods-controller.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+});
+};
+
+function update (request, response){
+  Food.find(request.params.id)
+  .then((data) => {
+    let food = data.rows[0]
+    let name = request.body.food_name
+    let calories = request.body.calories
+    console.log(food)
```
Comment (case-eee) [inline lib/controllers/foods-controller.js:69]: remove console.log's before pushing your code up

File: lib/controllers/foods-controller.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    Food.update(name, calories, food.id).then((data) => {
+      response.status(201).json(data)
+    });
+  });
+  };
+
+module.exports = {
```
Comment (case-eee) [inline lib/controllers/foods-controller.js:77]: 👍

File: lib/models/food.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+function update(food_name, calories, id){
+  return database.raw('UPDATE foods SET food_name=?, calories=? WHERE id =? RETURNING *', [food_name, calories, id])
+};
+
+function destroyAll () {
+  return database.raw('TRUNCATE foods RESTART IDENTITY')
+};
+
+
+module.exports = {
```
Comment (case-eee) [inline lib/models/food.js:35]: 👍

File: test/server-test.js [hunk 6]
```diff
@@ -15,131 +21,158 @@ describe('Server', function() {
...
-  //       done();
-  //     });
-  //   });
+  //   afterEach((done) => {
+  //     database.raw('TRUNCATE foods RESTART IDENTITY')
+  //     .then(() => done())
+  //   }); 
 
-  //   it('should return all of the food', function(done){
+  //   it ('return all items on the food table', (done) => {
```
Comment (case-eee) [inline test/server-test.js:70]: remove commented code before pushing things up!

**Review by case-eee — COMMENTED** [review]
@JonathanSR this is looking really great! i left ya some comments. in general, watch your spacing. also - you have functionality here that's not tested. be sure you're testing everything you possibly can - your FoodsController, your Food model, etc. good work 👍

**Pull request merged**